docs/adr/0001-pr-polling-workflow.md

```diff
@@ -0,0 +1,55 @@
+# 1. Pull Request Polling Workflow
+
+* Status: proposed
+* Date: 2023-06-20 
+* Authors: @yitsushi
+* Deciders: @yitsushi @chanwit @squaremo @yiannistri
+
+## Context
+
+To detect pull request changes, we can use webhooks or polling using GitHub's
+API.
+
+## Decision
+
+We decided to use polling for security reasons. Using webhook would require to
+open an ingress to the cluster and that's not acceptable especially in an
+air-gapped environment.
+
+The Branch-Based Planner has two components:
+
+1. Polling Server: Detect Pull Request changes and manage Teraform resource
+   state.
+2. Informer: Make a comment when new plan output is available.
+
+
+Full workflow and responsibilities:
```

Comment:
A lot of this is a design that would be relevant and accurate for webhooks, too. On the basis that an ADR should record one decision, perhaps this one should just explain the decision to use polling.